client/src/layouts/header/searchForm/Cities/Cities.jsx

```diff
@@ -1,15 +1,33 @@
 import React from 'react';
 import classes from './Cities.module.css';
 import City from './City/City';
+import { v4 as uuidv4 } from 'uuid';
+import PropTypes from 'prop-types'
+
+function Cities({ cities, setWeather, setLocation }) {
+  const onClickHandle = city => {
+    setWeather(city);
```

Comment:
If you find yourself doing same actions with the same data simultaneously like here (and I think I saw that in other places) - that may indicate that you didn't thought well through the state shape.